cmd/branchingconfigmanagers/tide-config-manager/main_test.go

```diff
@@ -41,25 +73,15 @@ func TestReconcile(t *testing.T) {
 				event:  featureFreezeEvent,
 				config: &prowconfig.ProwConfig{Tide: config.Tide{Queries: config.TideQueries{
 					{
-						Repos:            []string{"openshift/dummy"},
-						Labels:           []string{"qe-approved"},
-						IncludedBranches: []string{"master"},
+						Repos:            repos,
+						Labels:           []string{qeApproved},
+						IncludedBranches: []string{masterBranch},
 					},
 				}}},
 			},
 			wantErr: false,
 			expectedShardFiles: map[string]string{
-				"openshift/dummy/_prowconfig.yaml": `tide:
-  queries:
-  - includedBranches:
-    - master
-    labels:
-    - docs-approved
-    - px-approved
-    - qe-approved
-    repos:
-    - openshift/dummy
-`},
+				path: prepareProwConfig(repos, []string{docsApproved, pxApproved, qeApproved}, []string{masterBranch})},
```

Comment:
@droslean could not use here testhelper as the common function in pkg is not ready for it, and it would need further refactoring. Instead, I created a simple mechanism, not to paste configs every time. I think that it's enough, let me know if ok for you.

Reply:
Why the testhelper pkg is not ready for it? What it would take to make it ready?

Reply:
Bad working myabe. Look at `shardprowconfig.ShardProwConfig`. It's not exposing the data in any way. I could refactor it, but it is code-shared with other tool which does not take the benefit of `testhelper`. So, it would be refactoring of 2 tools just to make it available in tests.

Reply:
I think it is better to refactor the reconcile function. The fact that it is taking a target as an argument smells. In the tests, you could define the config and then after running reconcile you can check `testhelper.CompareWithFixture(t, tt.args.config)`. The expected config should exist in the `testdata` folder for better readability. Your code shouldn't walk in any directory, instead you can just load the prow config and loop through the shared configuration and reconcile them. Note that the code should also reconcile the non-sharded config as well.

Reply:
So to make what you want, I should refactor `shardprowconfig.ShardProwConfig` and determinize-prow-config tool and my new tool. I think it's slightly beyond the objective for this task.

Reply:
Nope, I inject additional operations to the `shardprowconfig.ShardProwConfig` so it has everything with the tool.

Reply:
Just to capture the explanation why `testhelper.CompareWithFixture` is not used in this particular place:
https://github.com/openshift/ci-tools/blob/88cd94e25cd685b4da4bd19dbb44aa7a8bd74010/pkg/api/shardprowconfig/shardprowconfig.go#L119
In the above place `shardprowconfig.ShardProwConfig` function cleans tide queries from the config which are needed to make the comparison.

Queries saving to the disk is done in the same function:
https://github.com/openshift/ci-tools/blob/88cd94e25cd685b4da4bd19dbb44aa7a8bd74010/pkg/api/shardprowconfig/shardprowconfig.go#L122
so changing that would mean refactor not only here, but also in the other tool which is using this function.

It was agreed that the above work would be extracted as a side-job to a separate card as to be done in the future as code cleaning/improvement.